lib/src/main/kotlin/com/swmansion/starknet/account/Account.kt

```diff
@@ -76,6 +77,23 @@ interface Account {
         forFeeEstimate: Boolean = false,
     ): DeclareTransactionPayload
 
+    /**
+     * Sign TypedData for off-chain usage with this account privateKey
+     *
+     * @param typedData a TypedData instance to sign
+     * @return a signature of typedData provided
+     */
+    fun signTypedData(typedData: TypedData): Signature
+
+    /**
+     * Verify a signature of TypedData on StarkNet
+     *
+     * @param typedData a TypedData instance which signature will be verified
+     * @param signature a signature of typedData
+     * @return `true` if signature is valid, `false` otherwise
+     */
+    fun verifyTypedDataSignature(typedData: TypedData, signature: Signature): Request<Boolean>
+
```

Comment:
why not just `sign` and `verify`?

Reply:
I wanted to make `signTypedData` a distinct method, similarly to `signDeclare` etc. `sign` is already used for InvokeTransactions.

`verifyTypedDataSignature` followed that convention.